src/payment/checkout/payment-form/flex-microform/CreditCardNumberField.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getCardIconForType } from '../utils/credit-card';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+
+import FlexMicroformField from './FlexMicroformField';
+
+class CreditCardNumberField extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            cardIcon: null,
+        }
+        this.onNumberChange = this.onNumberChange.bind(this);
+    }
+
+    onNumberChange(data) {
+      let cardIcon = null;
+      if (data.card && data.card.length > 0) {
+        cardIcon = getCardIconForType(data.card[0].cybsCardType);
+      }
+      this.setState({ cardIcon });
+    }
+
+    render() {
+        return (<div className="col-lg-6 form-group">
+            <FlexMicroformField
+                fieldType="number"
+                microform={this.props.microform}
+                disabled={this.props.disabled}
+                onChange={this.onNumberChange}
+                label={<FormattedMessage
```

Comment:
Is this handling of i18n correct?

Reply:
It's okay.  You're passing a component in and rendering it - it's not super canonical, but it works.  

I think the ideal form would be to use the `this.props.intl.formatMessage()` form of localization here, which will result in a string that you pass in, rather than a component.  It does mean you'll need a messages.js file to hold the messages.  It's better for passing the translated strings around as 'data' like this.